code/_onclick/item_attack.dm

```diff
@@ -57,136 +54,92 @@
-			if(attack_verb.len)
-				O.show_message("<span class='combat danger'>[M] has been [pick(attack_verb)] with [src][showname] </span>", 1)
-			else
-				O.show_message("<span class='combat danger'>[M] has been attacked with [src][showname] </span>", 1)
+	add_logs(user, M, "attacked", name, "(INTENT: [uppertext(user.a_intent)]) (DAMTYPE: [uppertext(damtype)])", print_attack_log = (force > 0))//print it if stuff deals damage
+	add_fingerprint(user)
 
-		if(!showname && user)
-			if(user.client)
-				to_chat(user, "<span class='combat danger'>You attack [M] with [src]. </span>")
 
+//the equivalent of the standard version of attack() but for object targets.
+/obj/item/proc/attack_obj(obj/O, mob/living/user)
+	if(flags & (NOBLUDGEON))
+		return
+	user.changeNext_move(CLICK_CD_MELEE)
+	user.do_attack_animation(O)
+	O.attacked_by(src, user)
 
+/atom/movable/proc/attacked_by()
+	return
 
-	if(istype(M, /mob/living/carbon/human))
-		return M:attacked_by(src, user, def_zone)	//make sure to return whether we have hit or miss
+/obj/attacked_by(obj/item/I, mob/living/user)
+	if(I.force)
+		user.visible_message("<span class='danger'>[user] has hit [src] with [I]!</span>", "<span class='danger'>You hit [src] with [I]!</span>")
+	take_damage(I.force, I.damtype, "melee", 1)
+
+/mob/living/attacked_by(obj/item/I, mob/living/user, def_zone)
+	send_item_attack_message(I, user)
+	if(I.force)
+		apply_damage(I.force, I.damtype, def_zone)
+		if(I.damtype == BRUTE)
+			if(prob(33))
+				I.add_mob_blood(src)
+				var/turf/location = get_turf(src)
+				add_splatter_floor(location)
+				if(get_dist(user, src) <= 1)	//people with TK won't get smeared with blood
+					user.add_mob_blood(src)
+		return TRUE //successful attack
+
+/mob/living/simple_animal/attacked_by(obj/item/I, mob/living/user)
+	if(!I.force)
+		user.visible_message("<span class='warning'>[user] gently taps [src] with [I].</span>",\
+						"<span class='warning'>This weapon is ineffective, it does no damage!</span>")
+	else if(I.force < force_threshold || I.damtype == STAMINA)
+		visible_message("<span class='warning'>[I] bounces harmlessly off of [src].</span>",\
+					"<span class='warning'>[I] bounces harmlessly off of [src]!</span>")
 	else
-		switch(damtype)
-			if("brute")
-				if(istype(src, /mob/living/carbon/slime))
-					M.adjustBrainLoss(power)
+		return ..()
 
-				else
+// Proximity_flag is 1 if this afterattack was called on something adjacent, in your square, or on your person.
+// Click parameters is the params string from byond Click() code, see that documentation.
+/obj/item/proc/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	return
 
-					M.take_organ_damage(power)
-					if(prob(33)) // Added blood for whacking non-humans too
-						M.add_splatter_floor()
-			if("fire")
-				M.take_organ_damage(0, power)
-				to_chat(M, "Aargh it burns!")
-		M.updatehealth()
-	add_fingerprint(user)
+/obj/item/proc/get_clamped_volume()
```

Comment:
Excellent; this system works a lot better for making more damaging items louder.